C++/Logger/conversion/Stringifiable.h

```diff
@@ -2,17 +2,20 @@
 #define _STRINGIFIABLE_H_
 
 #include "ConvertibleTo.h"
+#include "../metaprogramming/value_type_is.h"
 
 #include <string>
 
 namespace Toolbox::Log::Conversion
 {
 
-class Stringifiable : public ConvertibleTo<std::string>
+class Stringifiable
+  : public ConvertibleTo<std::string>
+  , value_type_is <std::string>
 {
 public:
   virtual const std::string toString (void) const = 0;
-  operator std::string (void) const override final;
```

Comment:
Its an implementation of an interface.  There's no definition.  This is not method overriding.  On gcc this does not compile.  
```
$ make
g++ --std=c++1z -c conversion/Stringifiable.cpp
In file included from conversion/Stringifiable.cpp:1:0:
conversion/Stringifiable.h:18:31: error: expected ‘;’ at end of member declaration
   operator std::string (void) override const;
                               ^~~~~~~~
conversion/Stringifiable.h:18:40: error: declaration does not declare anything [-fpermissive]
   operator std::string (void) override const;
                                        ^~~~~
conversion/Stringifiable.h:18:3: error: ‘Toolbox::Log::Conversion::Stringifiable::operator std::string()’ marked ‘override’, but does not override
   operator std::string (void) override const;
   ^~~~~~~~
conversion/Stringifiable.cpp:6:1: error: prototype for ‘Toolbox::Log::Conversion::Stringifiable::operator std::string() const’ does not match any in class ‘Toolbox::Log::Conversion::Stringifiable’
 Stringifiable::operator std::string (void) const
 ^~~~~~~~~~~~~
In file included from conversion/Stringifiable.cpp:1:0:
conversion/Stringifiable.h:18:3: error: candidate is: Toolbox::Log::Conversion::Stringifiable::operator std::string()
   operator std::string (void) override const;
   ^~~~~~~~
makefile:36: recipe for target 'Stringifiable.o' failed
make: *** [Stringifiable.o] Error 1
```